feat(descriptive error):create a userValidations.js

habibaudu opened the pull request:
#### What does this PR do?
Sends users descriptive Registration or signup validation error

#### Description of Task to be completed?
- Validate user registration
- Send a descriptive error message for each invalid input
- Remove duplicate email check from models
- Check for duplicate emails and send an appropriate descriptive message
- Create user.test.js file
- Write a test for all possible error message
- Comment out catchAllRoutes in app.js (it's not working and preventing other things from running)
Write a get catchall route in app.js(subject to change)
#### How should this be manually tested?

#### Any background context you want to provide?
Fire up postman
   - Enter registration url http://localhost:3000/api/v1/user
   - Enter invalid username, password or email.
To test
     On the Terminal run :  npm test
#### What are the relevant pivotal tracker stories?

           https://www.pivotaltracker.com/story/show/162727552

#### Screenshots (if appropriate):
   None
#### Questions:
      None

habibaudu committed:
feat(descriptive error):create a userValidations.js
validate user registration
send descriptive error message
check for duplicate emails and send appropriate descriptive message
create user.test.js file
write test for all possible error message
comment out catchAllRoutes in app.js (its not working and preventing other things from running)
write a get catchall route in app.js(subject to change)
[finish #162727552]

habibaudu committed:
feat(descriptive error):create a userValidations.js
validate user registration
send descriptive error message
check for duplicate emails and send appropriate descriptive message
create user.test.js file
write test for all possible error message
comment out catchAllRoutes in app.js (its not working and preventing other things from running)
write a get catchall route in app.js(subject to change)
[finish #162727552]

oreoluwade commented on app.js:
```diff
@@ -25,8 +25,9 @@ app.use(
   })
 );
-
+app.get('*', (req, res) => res.status(404).send(
```
You can take this catch all route handler out for now

habibaudu replied:
ok. will do that.

oreoluwade commented:
The relevant pivotal tracker story should be a link that takes the clicker directly to PT. Please edit that.

oreoluwade commented on middlewares/userValidations.js:
```diff
@@ -0,0 +1,100 @@
+ */
+class userValidations {
+  /**
+    * Controll update a user.
```
Is this comment meant for this method?

habibaudu replied:
I will fix that.Thanks

oreoluwade commented on middlewares/userValidations.js:
```diff
@@ -0,0 +1,100 @@
+    */
+  static signup(req, res, next) {
+    const usernamecharTest = /[^a-zA-Z/\s/-]/g;
+    const validEmail = /[a-zA-Z0-9]+@[a-z]+\.[com]+/;
```
emails don't have to end with `.com`

habibaudu replied:
ok. will take care of that right away

oreoluwade commented on middlewares/userValidations.js:
```diff
@@ -0,0 +1,100 @@
+import models from '../models';
+
+const { user } = models;
```
Find a way to ensure that the models always start with uppercase so that this won't be confusing. In essence, `User` instead of `user`

habibaudu replied:
ok. will do that

oreoluwade commented on middlewares/userValidations.js:
```diff
@@ -0,0 +1,100 @@
+              { message: 'Email already in Use, No duplicate emails' });
+          }
+        })
+        .catch(() => res.status(500).json(
```
These errors are supposed to be sent to the all catch error handler.

habibaudu replied:
ok.

habibaudu committed:
feat(descriptive error):fix validation errors
[finish #162727552]

habibaudu committed:
feat(descriptive error):fix validation errors
[finish #162727552]

habibaudu committed:
feat(register errors): install validator library
remove some test
[#162727552]

habibaudu committed:
feat():fix bug
[#16272552]

habibaudu committed:
feat()pulled from develop

habibaudu committed:
feat()pulled from develop

habibaudu committed:
fixing conflict

habibaudu committed:
fixing conflict

habibaudu committed:
[#162727552]

Commit:
add notFound route

Commit:
refactor file imports

habibaudu committed:
[#162727552] refactor routes

habibaudu committed:
[#162727552] refactor routes

habibaudu committed:
[#162727552] refactor user validations

habibaudu committed:
[#162727552] refactor user validations

habibaudu committed:
[#162727552] add undo migration script

habibaudu committed:
feat(descriptive errors): write passwordValidator for password validation
write usernameValidator for username validation
write emailIsValid middleware email validation
write usernameExistValidation  to check if username already exist
write emailExistValidation  to check if email already exist
[#162727552]

habibaudu committed:
feat(descriptive errors): installs babel-polyfill on dependencies
[#162727552]

habibaudu committed:
modified user.test.js file
[#162727552]

habibaudu committed:
resolve merge conflict

habibaudu committed:
feat(resgistration errors):write tests for errors
[finish #162727552]

habibaudu committed:
remove git markers'

habibaudu committed:
resolves conflict
[#162727552]

habibaudu committed:
[#162727552] add env to test script

habibaudu committed:
feat(registration errors): write more tests
[#162727552]
